payjoin-cli/src/db/v2.rs

```diff
@@ -146,61 +161,75 @@ impl SessionPersister for ReceiverPersister {
-        if let Some(existing) = recv_tree.get(key)? {
-            let mut wrapper: SessionWrapper<ReceiverSessionEvent> =
-                serde_json::from_slice(&existing).map_err(Error::Deserialize)?;
-            wrapper.completed_at = Some(SystemTime::now());
-            let value = serde_json::to_vec(&wrapper).map_err(Error::Serialize)?;
-            recv_tree.insert(key, value.as_slice())?;
-        }
-        recv_tree.flush()?;
+        let conn = self.db.get_connection()?;
+
+        conn.execute(
+            "UPDATE receive_sessions SET completed_at = ?1 WHERE session_id = ?2",
+            params![now(), self.session_id.as_integer()],
+        )?;
+
         Ok(())
     }
 }
 
 impl Database {
     pub(crate) fn get_recv_session_ids(&self) -> Result<Vec<SessionId>> {
-        let recv_tree = self.0.open_tree("recv_sessions")?;
+        let conn = self.get_connection()?;
+        let mut stmt =
+            conn.prepare("SELECT session_id FROM receive_sessions WHERE completed_at IS NULL")?;
+
+        let session_rows = stmt.query_map([], |row| {
+            let session_id: i64 = row.get(0)?;
+            Ok(SessionId::Receive(session_id))
+        })?;
+
         let mut session_ids = Vec::new();
-        for item in recv_tree.iter() {
-            let (key, value) = item?;
-            let wrapper: SessionWrapper<ReceiverSessionEvent> =
-                serde_json::from_slice(&value).map_err(Error::Deserialize)?;
-            if wrapper.completed_at.is_some() {
-                continue;
-            }
-            session_ids.push(SessionId::new(u64::from_be_bytes(
-                key.as_ref().try_into().map_err(Error::TryFromSlice)?,
-            )));
+        for session_row in session_rows {
+            let session_id = session_row?;
+            session_ids.push(session_id);
         }
+
         Ok(session_ids)
     }
 
     pub(crate) fn get_send_session_ids(&self) -> Result<Vec<SessionId>> {
-        let send_tree = self.0.open_tree("send_sessions")?;
+        let conn = self.get_connection()?;
+        let mut stmt =
+            conn.prepare("SELECT session_id FROM send_sessions WHERE completed_at IS NULL")?;
+
+        let session_rows = stmt.query_map([], |row| {
+            let session_id: i64 = row.get(0)?;
+            Ok(SessionId::Send(session_id))
+        })?;
+
         let mut session_ids = Vec::new();
-        for item in send_tree.iter() {
-            let (key, value) = item?;
-            let wrapper: SessionWrapper<SenderSessionEvent> =
-                serde_json::from_slice(&value).map_err(Error::Deserialize)?;
-            if wrapper.completed_at.is_some() {
-                continue;
-            }
-            session_ids.push(SessionId::new(u64::from_be_bytes(
-                key.as_ref().try_into().map_err(Error::TryFromSlice)?,
-            )));
+        for session_row in session_rows {
+            let session_id = session_row?;
+            session_ids.push(session_id);
         }
+
```

Comment:
Nit: noticing new lines added here and there. Please becareful to keep formatting changes outside of non formatting commits.